gulpfile.js

```diff
@@ -0,0 +1,142 @@
+var autoprefixer = require("gulp-autoprefixer");
+var changed = require("gulp-changed");
+var connect = require("gulp-connect");
+var eslint = require("gulp-eslint");
+var gulp = require("gulp");
+var gutil = require("gulp-util");
+var less = require("gulp-less");
+var minifyCSS = require("gulp-minify-css");
+var path = require("path");
+var uglify = require("gulp-uglify");
+var webpack = require("webpack");
+
+var dirs = {
+  src: "./src",
+  js: "./src/js",
+  dist: "dist",
+  styles: "./src/css",
+  img: "./src/img",
+  imgDist: "img"
+};
+
+var files = {
+  mainJs: "main",
+  mainJsDist: "main",
+  mainLess: "main",
+  mainCssDist: "main",
+  index: "index.html"
+};
+
+var webpackConfig = {
+  entry: dirs.js + "/" + files.mainJs + ".jsx",
+  output: {
+    path: path.resolve(dirs.dist),
+    filename: files.mainJsDist + ".js"
+  },
+  module: {
+    loaders: [
+      {
+        test: /\.jsx$/,
+        loader: "babel-loader",
+        exclude: /node_modules/
+      }
+    ],
+    postLoaders: [
+      {
+        loader: "transform?envify"
+      }
+    ]
+  },
+  resolve: {
+    extensions: ["", ".jsx", ".js"]
+  }
+};
+
+// Use webpack to compile jsx into js,
+gulp.task("webpack", function (callback) {
+  // Extend options with source mapping
+  if (process.env.GULP_ENV === "development" &&
+    !process.env.DISABLE_SOURCE_MAP ||
+    process.env.DISABLE_SOURCE_MAP === "false") {
+    webpackConfig.devtool = "source-map";
+    webpackConfig.module.preLoaders = [
+      {
+        test: /\.js$/,
+        loader: "source-map-loader",
+        exclude: /node_modules/
+      }
+    ];
+  }
+  // run webpack
+  webpack(webpackConfig, function (err) {
+    if (err) {
+      throw new gutil.PluginError("webpack", err);
+    }
+    callback();
+  });
+});
+
+gulp.task("eslint", function () {
+  return gulp.src([dirs.js + "/**/*.?(js|jsx)"])
+    .pipe(changed(dirs.dist))
+    .pipe(eslint())
+    .pipe(eslint.formatEach("stylish", process.stderr));
+});
+
+gulp.task("less", function () {
+  return gulp.src(dirs.styles + "/" + files.mainLess + ".less")
+    .pipe(changed(dirs.dist))
```

Comment:
Won't this cause for `less` to only get the files that change and there for potentially fail because it doesn't have certain variables? Depending on which file is saved?